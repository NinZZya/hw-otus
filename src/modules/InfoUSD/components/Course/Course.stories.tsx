import React from "react";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { Course, CourseProp } from "./Course";

export default {
  title: "Modules/InfoUSD/Components/Course",
  component: Course,
} as Meta;

export const WithValue: React.FC<CourseProp> = () => (
  <Course value={1} indicatorColor="#ff0000" />
);
