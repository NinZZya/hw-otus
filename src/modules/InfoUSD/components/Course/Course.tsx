import React from "react";
import { Card, Indicator, Label } from "../../../../componets";
import { getRandomColor } from "../../../../utils/utils";

export interface CourseProp {
  value: number;
  indicatorColor?: string;
}

export const Course: React.FC<CourseProp> = (props) => {
  const { value = 0, indicatorColor } = props;
  const color = indicatorColor ? indicatorColor : getRandomColor();

  return (
    <Card>
      <Indicator type={"card"} color={color} />
      <span>Current course:</span>
      <Label value={`$${value}`} />
    </Card>
  );
};
