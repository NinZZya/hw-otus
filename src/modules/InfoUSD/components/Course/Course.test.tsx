import React from "react";
import renderer from "react-test-renderer";
import { Course } from "./Course";

it("Course render coorectly", () => {
  const tree = renderer
    .create(<Course value={0} indicatorColor={"#000"} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
