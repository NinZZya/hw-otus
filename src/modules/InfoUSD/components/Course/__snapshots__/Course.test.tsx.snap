// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course render coorectly 1`] = `
<div
  className="css-1y7yulv"
>
  <span
    className="css-15rxe6g"
    color="#000"
  >
    â€¢
  </span>
  <span>
    Current course:
  </span>
  <span
    className="css-l3kone"
  >
    $0
  </span>
</div>
`;
